caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -40,24 +40,34 @@
 error = error_handler.get(log)
 
 
-@block(backend_type=TGISBackend.backend_type, base_module=PeftPromptTuning)
+@modules.module(backend_type=TGISBackend.backend_type, base_module=PeftPromptTuning)
 class PeftPromptTuningTGIS(ModuleBase):
     SUPPORTED_LOAD_BACKENDS = [TGISBackend.backend_type, backend_types.LOCAL]
     ## Module Interface ##
 
-    def __init__(self, base_model_name, prompt_cache_id, eos_token, verbalizer) -> None:
+    def __init__(
+            self,
+            base_model_name,
+            prompt_cache_id,
+            eos_token,
+            verbalizer,
+            enable_backend=True,
+            tgis_backend=None,
+        ) -> None:
         super().__init__()
         # Configure the internal client
-        self._client = module_backend_config.get_backend(
-            TGISBackend.backend_type
-        ).get_client(base_model_name)
+        if enable_backend:
```

Comment:
This is actually a workaround paradigm we are following until we decide on how "shared backend" stuff is going to play out finally in caikit.

Reply:
In what way though? Because if this is `False`, .run() fails due to an uninitialized client, right?

Reply:
yes, `enable_backend` is essentially meant to handle the case where we want to initialize the class but not run inference, for example, in case of model-bootstrap+conversion. Actually I added a note for this in other PR, I'll pull that over to here as well.

Also, I'll add a check with relevant error message in case this is empty and someone tries to use it for inferencing.